stores/metadata.go

```diff
@@ -1079,10 +1094,10 @@ END AS health`).
 		Joins("INNER JOIN sectors s ON s.db_slab_id = slabs.id").
 		Joins("LEFT JOIN contract_sectors se ON s.id = se.db_sector_id").
 		Joins("LEFT JOIN contracts c ON se.db_contract_id = c.id").
-		Joins("LEFT JOIN contract_set_contracts csc ON csc.db_contract_id = c.id").
-		Joins("LEFT JOIN contract_sets cs ON cs.id = csc.db_contract_set_id AND cs.name=?", set).
+		Joins("LEFT JOIN contract_set_contracts csc ON csc.db_contract_id = c.id AND csc.db_contract_set_id = slabs.db_contract_set_id").
+		Joins("LEFT JOIN contract_sets cs ON cs.id = csc.db_contract_set_id").
 		Group("slabs.id").
-		Having("health <= ?", healthCutoff).
+		Having("health <= ? AND slabs.db_contract_set_id = (SELECT id FROM contract_sets cs WHERE cs.name = ?)", healthCutoff, set).
```

Comment:
did you try and put this extra `name` condition elsewhere? It seems a bit odd to have it in the `HAVING` stage because you potentially include and thus calculate health for a huge amount of slabs in which you aren't interested

Reply:
hm ok, it's probably fine, I figured you'd have to update `TestUnhealthySlabsNoContracts`...

Reply:
Not the name but we only join on the corresponding set id. So in theory we already filter all links that don't belong to the set before the HAVING.